### PR TITLE
DAOS-9928 object: RPC & bulk hanlder for secondary provider

### PR DESCRIPTION
**Pull request by NiuYawei**
RPC & bulk handler for secondary provider.

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### CODE REVIEW

**Comment by NiuYawei** [comment]
Pushed as draft since the Cart changes are not landed in feature branch yet. Alex, could you take a look to see if the Cart APIs are properly used? There is a "TODO" comment about how to handle the forwarded RPC, do you have any thoughts over that? Thanks in advance.

**Comment by NiuYawei** [comment]
Rebased to pick up Alex's first Cart patch.

File: src/engine/srv.c [hunk 1]
```diff
@@ -296,6 +296,45 @@ dss_rpc_hdlr(crt_context_t *ctx, void *hdlr_arg,
...
+	int			 xs_id, rc;
+
+	D_DEBUG(DB_TRACE, "Received secondary RPC, opc: %#x\n", rpc->cr_opc);
+
+	rc = crt_req_dst_tag_get(rpc, &tag);
+	if (rc) {
+		D_ERROR("Failed to get tag from RPC, "DF_RC"\n", DP_RC(rc));
+		return rc;
+	}
+
```
Comment (frostedcmos) [inline src/engine/srv.c:314]: might be good to print here tag as debug before starting processing/error checking

Comment (NiuYawei) [reply src/engine/srv.c:314]: Sure, I'll add it in follow-on patch.

**Review by frostedcmos — APPROVED** [review]
cart related parts lgtm

**Review by liuxuezhao — APPROVED** [review]

File: src/engine/srv.c [hunk 2]
```diff
@@ -462,10 +501,7 @@ dss_srv_handler(void *arg)
 
 	xs_type = xs_id2type(dx->dx_xs_id);
 	if (xs_type == DSS_XS_SEC) {
-		/* TODO: Create secondary cart context and register secondary RPC handler */
-		D_ASSERTF(0, "Secondary cart context isn't supported\n");
-#if 0
-		rc = crt_context_create_seconary(&dmi->dmi_ctxt);
+		rc = crt_context_create_secondary(&dmi->dmi_ctx, 0);
```
[the remark this answers is not in the document]
Comment (NiuYawei) [reply src/engine/srv.c:504]: yes, only 0 is supported so far.

**Pull request merged**